src/Python/qsharp-core/qsharp/loader.py

```diff
@@ -122,13 +120,77 @@ def simulate_noise(self, **kwargs) -> Any:
         """
         return qsharp.client.simulate_noise(self, **kwargs)
 
-    def as_qir(self) -> bytes:
+    def as_qir(self, metadata: Optional[Dict[str, str]] = None, **kwargs) -> str:
```

Comment:
I think it would be good to keep this simple and just get `target` and `target_capability` from `kwargs` instead of also having a `metadata` argument.

Reply:
As discussed, I removed the `metadata` parameter in favor of just the `kwargs`.